codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -236,7 +236,8 @@ private void writePendingBytes() {
 
     /**
      * This will allocate bytes by stream weight and priority for the entire tree rooted at {@code parent}, but does
-     * not write any bytes.
+     * not write any bytes. Priority is generally distributed proportionally to siblings in the priority tree, however
+     * we ensure that at least 1 byte is assigned to each stream in order to make progress.
```

Comment:
The "progress" statement was in reference to ensuring that we allocate all of the connection window for a given call to `allocateBytesForTree`.  This is to pick up the slack if `connectionWindow * weightRatio` value happens to be `< 1` for all siblings.  @buchgr - Can you explain the rational behind allocating `a few hundred bytes`?  This amount seems a bit arbitrary to me and how would we ensure this amount is distributed proportionally to weight/priority?

Leave the `Math.Max()` to ensure progress is made and all connection window is distributed.

Change the comment you added to reflect something like the following:
> We should be ensuring that the entire connection window is used (assuming streams have >= connection window bytes to send) and we may need some sort of rounding to accomplish this.